packages/client/wallets/aa/src/blockchain/wallets/eoa.ts

```diff
@@ -0,0 +1,60 @@
+import { CrossmintWalletService } from "@/api";
+import { EVMAAWallet } from "@/blockchain";
+import type { UserIdentifier, WalletConfig } from "@/types";
+import { CURRENT_VERSION, ZERO_DEV_TYPE, createOwnerSigner } from "@/utils";
+import { signerToEcdsaValidator } from "@zerodev/ecdsa-validator";
+import { createKernelAccount } from "@zerodev/sdk";
+import { EntryPoint, EntryPointVersion } from "permissionless/types/entrypoint";
+import { HttpTransport, PublicClient } from "viem";
+
+import { EVMBlockchainIncludingTestnet, blockchainToChainId } from "@crossmint/common-sdk-base";
+
+export default class EOAWalletService {
+    constructor(private readonly crossmintService: CrossmintWalletService) {}
+
+    public async getOrCreate({
+        userIdentifier,
+        chain,
+        publicClient,
+        entryPoint,
+        entryPointVersion,
+        walletConfig,
+    }: {
+        userIdentifier: UserIdentifier;
+        chain: EVMBlockchainIncludingTestnet;
+        publicClient: PublicClient<HttpTransport>;
+        entryPoint: EntryPoint;
+        entryPointVersion: EntryPointVersion;
+        walletConfig: WalletConfig;
+    }) {
+        const eoa = await createOwnerSigner({
+            chain,
+            walletConfig,
+        });
+        const ecdsaValidator = await signerToEcdsaValidator(publicClient, {
+            signer: eoa,
+            entryPoint,
+        });
+        const account = await createKernelAccount(publicClient, {
+            plugins: {
+                sudo: ecdsaValidator,
+            },
+            index: BigInt(0),
+            entryPoint,
+        });
+
+        const wallet = new EVMAAWallet(account, this.crossmintService, publicClient, entryPoint, chain);
+        await this.crossmintService.storeAbstractWallet({
+            userIdentifier,
+            type: ZERO_DEV_TYPE,
+            smartContractWalletAddress: account.address,
+            eoaAddress: eoa.address,
```

Comment:
This is outdated, we should send something like this: https://github.com/Crossmint/crossmint-sdk/pull/488/files#diff-54c932ab0149942d2ac2dcc04b35da86081332beee12815702d8c02bdeba601fR54

With this new format: https://github.com/Crossmint/crossmint-sdk/pull/488/files#diff-c0a3c29265bf1b60a9a23113ccf53283a5864674ea6bc39e03a40d02515b575aR11

```
await this.crossmintService.storeAbstractWallet({
            userIdentifier,
            type: ZERO_DEV_TYPE,
            smartContractWalletAddress: account.address,
            signerData: {
                eoaAddress: eoa.address,
                type: "eoa",
            },
            version: CURRENT_VERSION,
            baseLayer: "evm",
            chainId: blockchainToChainId(chain),
            entryPointVersion,
        });
```
We can merge this if not and I'll submit a PR with those changes to main. Was pushing it in the other branch

Reply:
I mean, it will technically work as the endpoint is backwards compatible in crossbit. But let's adopt the new format here while we are at it

Reply:
ofc, forgot about that, will update